ops/lib.rs

```diff
@@ -452,7 +452,9 @@ fn codegen_u8_slice(core: &TokenStream2, idx: usize) -> TokenStream2 {
       Ok(b) => {
         // Handles detached buffers.
         let byte_length = b.byte_length();
-        let store = b.data() as *mut u8;
+        let data = b.data();
+        assert!(data.is_some());
+        let store = data.unwrap().cast::<u8>().as_ptr();
```

Comment:
Changes in this file are related to https://github.com/denoland/rusty_v8/pull/1131

CC @andreubotella @littledivy

Reply:
This is related to #16161. IIUC `ArrayBuffer::data` will return `None` for all empty buffers, in which case most APIs would expect `&[]` or `&mut []`.

For FFI APIs, however, this might break things, and it might need the `#[op]` macro to also support `Option<&[u8]>`.

Reply:
Thanks for the heads up, I'm gonna merge the PR if it goes green and let Divy handle that in a follow up.

Reply:
Welp, it seems I hit that problem already in `test_ffi/tests/test.js` in 
```
const emptyBuffer = new BigUint64Array(0);
console.log(Boolean(dylib.symbols.is_null_ptr(Deno.UnsafePointer.of(emptyBuffer))));
```
It prints `false` while it should print `true`

Reply:
Not really. Only the ptr_of op is probably interested in the real pointer behind the slice instead of being interested in the slice itself.

There's just no good way to (currently at least) explain this to serde_v8 since while a plain pointer would be accurate to what the op needs, it doesn't really explain the JS param type.

Reply:
My idea was to that op parameters could be either `&[u8]` or `Option<&[u8]>`, so that only ops that care about the pointer possibly being null have to deal with this.

Reply:
Thanks for comments. it's a bit unclear what to do here, I'll wait for Divy's opinion

Reply:
`Option<&[u8]>` interfers with `undefined` and `null` values in serde_v8. I opened a PR adding support for raw ptr in ops macro: https://github.com/denoland/deno/pull/16826